src/cluster-legacy.c

```diff
@@ -5901,87 +5655,236 @@ sds genClusterInfoString(void) {
-"FLUSHSLOTS",
-"    Delete current node own slots information.",
-"INFO",
-"    Return information about the cluster.",
-"KEYSLOT <key>",
-"    Return the hash slot for <key>.",
-"MEET <ip> <port> [<bus-port>]",
-"    Connect nodes into a working cluster.",
-"MYID",
-"    Return the node id.",
-"MYSHARDID",
-"    Return the node's shard id.",
-"NODES",
-"    Return cluster configuration seen by node. Output format:",
-"    <id> <ip:port@bus-port[,hostname]> <flags> <master> <pings> <pongs> <epoch> <link> <slot> ...",
-"REPLICATE <node-id>",
-"    Configure current node as replica to <node-id>.",
-"RESET [HARD|SOFT]",
-"    Reset current node (default: soft).",
-"SET-CONFIG-EPOCH <epoch>",
-"    Set config epoch of current node.",
-"SETSLOT <slot> (IMPORTING <node-id>|MIGRATING <node-id>|STABLE|NODE <node-id>)",
-"    Set slot state.",
-"REPLICAS <node-id>",
-"    Return <node-id> replicas.",
-"SAVECONFIG",
-"    Force saving cluster configuration on disk.",
-"SLOTS",
-"    Return information about slots range mappings. Each range is made of:",
-"    start, end, master and replicas IP addresses, ports and ids",
-"SHARDS",
-"    Return information about slot range mappings and the nodes associated with them.",
-"LINKS",
-"    Return information about all network links between this node and its peers.",
-"    Output format is an array where each array element is a map containing attributes of a link",
-NULL
-        };
-        addReplyHelp(c, help);
-    } else if (!strcasecmp(c->argv[1]->ptr,"meet") && (c->argc == 4 || c->argc == 5)) {
+    /* Terminate the link based on the direction or all. */
+    if (!strcasecmp(c->argv[3]->ptr, "from")) {
+        freeClusterLink(n->inbound_link);
+    } else if (!strcasecmp(c->argv[3]->ptr, "to")) {
+        freeClusterLink(n->link);
+    } else if (!strcasecmp(c->argv[3]->ptr, "all")) {
+        freeClusterLink(n->link);
+        freeClusterLink(n->inbound_link);
+    } else {
+        addReplyErrorFormat(c, "Unknown direction %s", (char *) c->argv[3]->ptr);
+    }
+    addReply(c, shared.ok);
+
+    return 1;
+}
+
+int clusterNodeConfirmedReachable(clusterNode  *node) {
+    return !(node->flags & (CLUSTER_NODE_NOADDR|CLUSTER_NODE_HANDSHAKE));
+}
+
+char *clusterNodeIp(clusterNode *node) {
+    return node->ip;
+}
+
+int clusterNodeIsSlave(clusterNode *node) {
+    return !clusterNodeIsMaster(node);
+}
+
+clusterNode *clusterNodeGetSlaveof(clusterNode *node) {
+    return node->slaveof;
+}
+
+char *clusterNodeGetName(clusterNode *node) {
+    return node->name;
+}
+
+int clusterNodeTimedOut(clusterNode *node) {
+    return nodeTimedOut(node);
+}
+
+int clusterNodeIsFailing(clusterNode *node) {
+    return nodeFailed(node);
+}
+
+int clusterNodeIsNoFailover(clusterNode *node) {
+    return node->flags & CLUSTER_NODE_NOFAILOVER;
+}
+
+char *clusterNodeGetShardId(clusterNode *node) {
+    return node->shard_id;
+}
+
+int clusterCommandSpecial(client *c) {
```

Comment:
Let's document the return code if this is changing.

Reply:
added documentation in cluster.h